polkadot/node/core/backing/src/lib.rs

```diff
@@ -1247,28 +1209,28 @@ async fn construct_per_relay_parent_state<Context>(
 
 enum SecondingAllowed {
 	No,
-	Yes(Vec<(Hash, Vec<usize>)>),
+	// On which leaves is seconding allowed.
+	Yes(Vec<Hash>),
```

Comment:
we can't always know the depth, because we may second a candidate for which we don't have the parent.
so we only limit the total number. we can no longer enforce one per depth